tests/core/nogo/coverage/coverage_cgo_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coverage_cgo_test
+
+import (
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "nogo")
+
+go_library(
+    name = "foo",
+	cgo = True,
+    srcs = ["foo.go"],
+    importpath = "foo"
+)
+
+go_test(
+    name = "foo_test",
+    srcs = ["foo_test.go"],
+    embed = [":foo"]
+)
+
+nogo(
+    name = "nogo",
+    deps = ["//noinit"],
+    visibility = ["//visibility:public"],
+)
+-- foo.go --
+package foo
+
+import "C"
+
+func Foo() string {
+	return "foo"
+}
+-- foo_test.go --
+package foo
+
+import "testing"
+
+func TestFoo(t *testing.T) {
+	if actual, expected := Foo(), "foo"; actual != expected {
+		t.Errorf("Foo() should return foo")
+	}
+}
+-- noinit/BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "noinit",
+    srcs = ["analyzer.go"],
+    importpath = "noinit",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@org_golang_x_tools//go/analysis",
+    ],
+)
+-- noinit/analyzer.go --
+package noinit
+import (
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const Name = "gochecknoinits"
+
+var Analyzer = &analysis.Analyzer{
+	Name: Name,
+	Doc:  "Checks that no init functions are present in Go cod",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, f := range pass.Files {
+		for _, decl := range f.Decls {
+			funcDecl, ok := decl.(*ast.FuncDecl)
+			if !ok {
+				continue
+			}
+			name := funcDecl.Name.Name
+			if name == "init" && funcDecl.Recv.NumFields() == 0 {
+				pass.Report(analysis.Diagnostic{
+					Pos:      funcDecl.Pos(),
+					Message:  fmt.Sprintf("don't use init function"),
+					Category: Name,
+				})
+			}
+		}
+	}
+
+	return nil, nil
+}
+`,
+		Nogo: `@//:nogo`,
+	})
+}
+
+func TestCoverageWithNogo(t *testing.T) {
```

Comment:
Let's rename this func to be specific to nogo + coverage.

Reply:
I suspect this is why the Windows test was failing btw. There is no Bazel sandbox on Windows so targets with similar name could collide.